state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1200,36 +1200,49 @@ impl<T: Config<I>, I: 'static> EgressApi<T::TargetChain> for Pallet<T, I> {
 			*id
 		});
 		let egress_id = (<T as Config<I>>::TargetChain::get(), egress_counter);
-		match maybe_ccm_with_gas_budget {
+		let egress_fee = match maybe_ccm_with_gas_budget {
 			Some((
 				CcmDepositMetadata { source_chain, source_address, channel_metadata },
 				gas_budget,
-			)) => ScheduledEgressCcm::<T, I>::append(CrossChainMessage {
-				egress_id,
-				asset,
-				amount,
-				destination_address: destination_address.clone(),
-				message: channel_metadata.message,
-				cf_parameters: channel_metadata.cf_parameters,
-				source_chain,
-				source_address,
-				gas_budget,
-			}),
-			None => ScheduledEgressFetchOrTransfer::<T, I>::append(FetchOrTransfer::<
-				T::TargetChain,
-			>::Transfer {
-				asset,
-				destination_address: destination_address.clone(),
-				amount,
-				egress_id,
-			}),
-		}
+			)) => {
+				ScheduledEgressCcm::<T, I>::append(CrossChainMessage {
+					egress_id,
+					asset,
+					amount,
+					destination_address: destination_address.clone(),
+					message: channel_metadata.message,
+					cf_parameters: channel_metadata.cf_parameters,
+					source_chain,
+					source_address,
+					gas_budget,
+				});
+				gas_budget
+			},
+			None => {
+				let (amount_after_fees, egress_fee) =
+					Self::withhold_transaction_fee(IngressOrEgress::Egress, asset, amount);
```

Comment:
would perhaps make it easier to see the order is correct, and harder to make a mistake if we returned a struct instead of a tuple from `withhold_transaction_fee` so we get named fields... especially because the types are the same